src/app/core/mods/showResources/resources.ts

```diff
@@ -0,0 +1,105 @@
+export class Resources {
+    private name: string;
+    private typeId: number;
+    private resources: {elemId: number, state: State}[] = [];
+
+    constructor(resource: Resource) {
+        this.name = resource.name;
+        this.typeId = resource.elementTypeId;
+        this.addOrUpdateResource(resource);
+    }
+
+    public addOrUpdateResource(resource: Resource) {
+        if (this.resources.length > 0 && this.resources.find(r => r.elemId == resource.elementId) != undefined) {
+            this.resources.find(r => r.elemId == resource.elementId).state = resource.elementState;
+        } else {
+            this.resources.push({elemId: resource.elementId, state: resource.elementState});
+        }
+    }
+
+    public getName(): string {
```

Comment:
name seems to be a constant, so need to use a getter : instead, make the property public and readonly